### DISCUSSION TITLE
File system routes precedence

### DISCUSSION THREAD

**Discussion by bernharduw** [gatsbyjs/gatsby#31805] [Help]
Hey, I'm wondering if there is a defined precedence for file-based routes.

In development, when I have a route in place like e.g. `{SanityPage__slug}.js` and add a static file `pages/articles.js` it renders that file even if there is one with a slug called "articles". I'm using this to have special pages that have special treatment, while still pulling in the content from the source.

However in production, the "articles" page is overwritten by the parameterized general page. This sometimes also happens when I restart development. I haven't found any pattern yet, but my expectation was that more specific paths override more general ones.

Is there a way to define the priority of which pages are used? I was wondering if reordering some plugins might change the precedence, but since there is no explicit plugin for file system routes it wouldn't actually make sense.

**Comment by LekoArts** [comment]
> I'm using this to have special pages that have special treatment, while still pulling in the content from the source.

This sounds like you should put that special logic inside `{SanityPage__slug}.js` itself.

Having two pages with the same path will always be problematic, it should be unique.

**Reply by bernharduw** [reply]
I understand - it basically means we need a single file in /pages and another folder structure that renders the actual page types.

I'm trying to move our projects from the createPages API to file-based routes. When using createPages we could override particular paths by placing a file into /pages. I was hoping that file system routes would simplify our code.

What do you think about defining the precedence like Next.js does it (https://nextjs.org/docs/routing/dynamic-routes#caveats)? It's deterministic and simple.
